Adafruit_AS7341.cpp

```diff
@@ -913,3 +933,196 @@ void Adafruit_AS7341::writeRegister(byte addr, byte val) {
+	SMUXPixelRegistersLocal[writeAddress] |= newBitMask; 	//Apply new value
+	
+	return true;
+}
+
+/**
+ * @brief Use a 32-bit config word to set all the diodes connected to an ADC channel
+ * Bits in the config word correspond to indexes of the diodes per the extra header file
+ * WARNING: INDEXES ARE ARBITRARY, AND NOT THE SAME AS PIXEL NUMBERS
+ * The config word is intended to allow easy reading of a configuration from a file
+ * @param inADC
+ * @param wordIn
+ */
+bool Adafruit_AS7341::setSMUXADCFromConfigWord(byte inADC, uint32_t wordIn) {
+	if(inADC > 6)	//Error checking
+		return 0;
+		
+	// Serial.println(wordIn,BIN);	
+	uint32_t bitMask = 1;
+	for(byte i=0; i<NUM_PIXELS; i++)
+	{
+		// Serial.println(bitMask,BIN);
+		if(wordIn & bitMask)	//That's a bitwiseAND, so only the relevant bit is checked
+		{
+			updateSMUXEntryLocal(i, inADC);
+			// Serial.println(bitMask,BIN);
+		}
+		bitMask = bitMask << 1;		
+	}
+	return true;
+}
+
+/**
+ * @brief Add a diode/pixel to a config word that is being generated
+ *
+ * @param inputWord (note it's a pointer)
+ * @param inputPixel (use a macro, like "F1_LEFT" or "PIXEL3")
+ */
+bool Adafruit_AS7341::addSMUXPixelToADCword(uint32_t* inputWord, byte inputPixel) {
+	if(inputPixel >= NUM_PIXELS)
+		return false;
+	
+	uint32_t bitMask = 1 << inputPixel;
+	*inputWord |= bitMask;
+	
+	return true;
+}
+
+/**
+ * @brief Remove a diode/pixel from a config word that is being generated
+ *
+ * @param inputWord (note it's a pointer)
+ * @param inputPixel (use a macro, like "F1_LEFT" or "PIXEL3")
+ */
+bool Adafruit_AS7341::removeSMUXPixelFromADCword(uint32_t* inputWord, byte inputPixel) {
+	if(inputPixel >= NUM_PIXELS)
+		return false;
+	
+	uint32_t bitMask = 1 << inputPixel;
+	bitMask = ~bitMask;	//Invert it for removing
+	*inputWord &= bitMask;
+	
+	return true;
+}
+
+/**
+ * @brief Generate a config word from the local config for an ADC channel
+ *
+ * @param inputPixel (use a macro, like "ADC0")
+ */
+uint32_t Adafruit_AS7341::getSMUXConfigWordForADC(byte inADC) {
+	if(inADC > 6)	//Error checking
+		return 0;
+		
+	byte addr = 0;
+	byte pos = 0;
+	byte bitMask = 0;
+	byte addrContents = 0;
+	uint32_t newConfigWord = 0;
+	for(byte i=0; i<NUM_PIXELS; i++) {
+		addr = pixelAddr[i];
+		pos = pixelPos[i];
+		addrContents = SMUXPixelRegistersLocal[addr];
+		
+		bitMask = inADC;
+		// bitMask = bitMask << (pixelPos[i] - 2);	//Shift 0 or 4 places	
+		
+		if(pixelPos[i] == 2) {
+			addrContents = addrContents << 4;	//Remove other value, if present
+			bitMask = bitMask << 4;				//Align with new addrContents
+		}
+		if(pixelPos[i] == 6) {
+			addrContents = addrContents >> 4;	//Remove other value, if present
+			// bitMask = bitMask  4;
+		}
+		if(addrContents == bitMask)
+			addSMUXPixelToADCword(&newConfigWord,i);		
+	}
+	return newConfigWord;	
+}
+
+/**
+ * @brief Debugging function, prints the local custom SMUX configuration
+ *
+ * @param inputPixel (use a macro, like "ADC0")
```

Comment:
This and the rest of the documentation blocks will need to be updated before merge